### PR TITLE
Feature/dao haus connect

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue
https://github.com/HausDAO/daohaus-monorepo/issues/120
https://github.com/HausDAO/daohaus-monorepo/issues/121

## Changes

- DaoHaus Connect Context
   - Starts with Quiver code
   - Removes excess functionality
   - Provides defaults to context. No props needed. 
   - Loads Profile
   - Determines correct network
   - Removes auto-switch functionality
   - Prevents diconnect on network of address change
   - Adds auto switch functionality 
   - Refactors all defaults and functionality to utils to maintain readability. 
   - Generates network typings around available networks passed in through context props.
   - chainId is a valid network Id type instead of a string type. 
   - Expands keychain to handle all 
   
- ConnetedUI 
    - Display when Connect button not connected
    - Displays loading (Spinner coming with atom) 
    - Displays connected state (User profile data, backups if no profile found, network data, user data) 
  
- NetworkUI
   - Dropdown for when out of DAOcontext (ex Hub or Summon) and not on valid network 
   - Dropdown renders buttons based on available networks in context 
   - Dropdown switcher UI will prompt MM switch to network when pressed
   - Button for when in DAOcontext to switch to DAO's network
   - Does not display when on correct network
   
- DaoHausNav
  - Packages and styles a container around ConnectedUI and NetworkUI 

## Packages Added

   "@walletconnect/ethereum-provider": "^1.7.8",
    "@walletconnect/web3-provider": "^1.7.8",
 "web3modal": "^1.9.7"
## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: libs/common-utilities/src/constants/endpoints.ts [hunk 1]
```diff
@@ -5,6 +5,22 @@ export const ENDPOINTS: KeychainList = {
     '0x4': 'https://api.thegraph.com/subgraphs/name/hausdao/daohaus-v3-rinkeby',
     '0x2a': 'https://api.thegraph.com/subgraphs/name/hausdao/daohaus-v3-kovan',
   },
+  // RPC: {
```
Comment (jordanlesich) [inline libs/common-utilities/src/constants/endpoints.ts:8]: Commenting out until process errors are fixed. Loacing RPCs in Connect package currently

File: libs/daohaus-connect-feature/src/HausConnectContext.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  const isConnected = useMemo(
+    () => !!provider && !!address && !!chainId,
+    [provider, address, chainId]
+  );
+  const isMetamask = useMemo(() => isMetamaskProvider(provider), [provider]);
+  const validNetwork = useMemo(
+    () => !!chainId && isValidNetwork(chainId, networks),
+    [chainId, networks]
+  );
+  // TODO, detect if we're in DAOscope by looking at the url.
```
Comment (jordanlesich) [inline libs/daohaus-connect-feature/src/HausConnectContext.tsx:84]: No way of deriving DAO scope until we start doing work on Core-UI or DAO context

File: libs/daohaus-connect-feature/src/components/NetworkButton/NetworkButton.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  getNetworkName,
+  Keychain,
+  networkData,
+  NetworkType,
+} from '@daohaus/common-utilities';
+import { Button, Dropdown, DropdownItem, ParXs } from '@daohaus/ui';
+import { BiError } from 'react-icons/bi';
+import { useHausConnect } from '../../HausConnectContext';
+
+//  Refactor
```
Comment (jordanlesich) [inline libs/daohaus-connect-feature/src/components/NetworkButton/NetworkButton.tsx:11]: I think this is artifact. I'll remove this

File: libs/daohaus-connect-feature/src/utils/contextHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+export const isMetamaskProvider = (
+  provider: providers.Web3Provider | undefined | null
+) => provider?.connection?.url === 'metamask';
+
+export const handleSetProvider = async ({
+  provider,
+  setWalletState,
+}: {
+  // eslint-disable-next-line
+  provider: any;
```
Comment (jordanlesich) [inline libs/daohaus-connect-feature/src/utils/contextHelpers.ts:32]: As discussed in Dev meeting.

File: libs/daohaus-connect-feature/src/utils/defaults.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { TEMPORARY_RPC } from './common';
+
+export const supportedNetworks = addKeychain(
+  ENDPOINTS.EXPLORER,
+  'explorer',
+  addKeychain(TEMPORARY_RPC, 'rpc')
+);
+
+const providerOptions: IProviderOptions = {
+  // authereum: {
```
Comment (jordanlesich) [inline libs/daohaus-connect-feature/src/utils/defaults.ts:13]: I can remove comments when we set up other networks

File: libs/ui/src/components/atoms/button/button.tsx [hunk 6]
```diff
@@ -15,8 +15,10 @@ export type ButtonProps = {
   leftAlign?: boolean;
   avatar?: boolean;
   onClick?: React.MouseEventHandler<HTMLButtonElement>;
-  icon?: IconType;
-  iconPos?: 'left' | 'right';
+  IconLeft?: IconType;
+  IconRight?: IconType;
+  // icon?: IconType;
```
Comment (jordanlesich) [inline libs/ui/src/components/atoms/button/button.tsx:20]: I'll delete

File: libs/common-utilities/src/constants/endpoints.ts (hunk 1 above)
Comment (skuhlmann) [reply libs/common-utilities/src/constants/endpoints.ts:8]: Let's create a new ticket to build this out as an input. 

Users of the package will need to pass the rpc.

File: libs/dao-data/.env.sample [hunk 7]
```diff
@@ -1 +1 @@
-NX_RIVET_KEY=
\ No newline at end of file
+NX_RIVET_KEY=
```
Comment (skuhlmann) [inline libs/dao-data/.env.sample:1]: Remove this.

**Review by skuhlmann — APPROVED** [review]
Approving with the changes noted below.

File: libs/common-utilities/src/constants/endpoints.ts (hunk 1 above)
Comment (alexkeating) [reply libs/common-utilities/src/constants/endpoints.ts:8]: I am leaning towards hardcoding these

**Pull request merged**